uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar.component.js

```diff
@@ -0,0 +1,351 @@
+      ])
+      .then(function (results) {
+        vm.supportData.absenceTypes = results[0];
+        vm.supportData.publicHolidays = results[1];
+        vm.supportData.leaveRequestStatuses = results[2].hrleaveandabsences_leave_request_status;
+        vm.supportData.dayTypes = results[2].hrleaveandabsences_leave_request_day_type;
+      });
+    }
+
+    /**
+     * Returns the structure of the month of the given date
+     *
+     * @param  {Object} dateMoment
+     * @return {Object}
+     */
+    function monthStructure (dateMoment) {
+      return {
+        index: dateMoment.month(),
+        year: dateMoment.year(),
+        name: dateMoment.format('MMM')
+      };
+    }
+
+    /**
+     * Reloads the selected months data
+     *
+     * If the source of the refresh is a period change, then
+     * it rebuilds the months list as well
+     * If the source of the refresh is a change in contacts filters, then
+     * it reloads the contacts as well
+     *
+     * @param {string} source The source of the refresh (period or contacts change)
+     */
+    function refresh (source) {
+      source = _.includes(['contacts', 'period'], source) ? source : 'period';
+
+      $q.resolve()
+        .then((source === 'period' ? buildPeriodMonthsList : _.noop))
+        .then((source === 'contacts' ? loadContacts : _.noop))
+        .then(function () {
+          injectAndShowMonths((source === 'contacts'));
+        });
+    }
+
+    /**
+     * Sends the "show" signal to the leave-calendar-month components
+     *
+     * @param {Boolean} forceDataReload if true, then a month will load its data
+     *   regardless if it had already loaded it
+     */
+    function sendShowMonthsSignal (forceDataReload) {
+      var monthsToShow = !vm.selectedMonths.length
+        ? vm.months
+        : vm.months.filter(function (month) {
+          return _.includes(vm.selectedMonths, month.index);
+        });
+
+      $rootScope.$emit('LeaveCalendar::showMonths', monthsToShow, !!forceDataReload);
+    }
+
+    /**
+     * Chooses the months that are to be selected by default
+     */
+    function setDefaultMonths () {
+      vm.selectedMonths = [moment().month()];
+    }
+
+    /**
+     * Sets the user's role based on his permissions
+     */
+    function setUserRole () {
+      if (vm.roleOverride) {
+        return $q.resolve().then(function () {
+          userRole = vm.roleOverride;
+        });
+      } else {
+        return $q.all([
+          checkPermissions(sharedSettings.permissions.admin.administer),
+          checkPermissions(sharedSettings.permissions.ssp.manage)
+        ])
+        .then(function (results) {
+          userRole = results[0] ? 'admin' : (results[1] ? 'manager' : 'staff');
+        });
+      }
+    }
+
+    /**
+     * Waits for the next digest cycle to make sure that all the data
+     * from the component had been transmitted to the child components
+     */
+    function waitForNextDigest () {
```

Comment:
I improved the logic now and got rid of the dependency on the angular digest cycle